index.js

```diff
@@ -1,56 +1,90 @@
 const FILE = 'pokedex.json';
 
-/**
- * ===================================
- * Configurations and set up
- * ===================================
- */
-
 // Init express app
 const app = express();
 
-/**
- * ===================================
- * Routes
- * ===================================
- */
+//<<REACT ENGINE>>
+const reactEngine = require('express-react-views').createEngine();
+app.engine('jsx', reactEngine);
+app.set('views', __dirname + '/views');
+app.set('view engine', 'jsx');
 
-app.get('/:id', (request, response) => {
-
-  // get json from specified file
-  jsonfile.readFile(FILE, (err, obj) => {
-    // obj is the object from the pokedex json file
-    // extract input data from request
-    let inputId = request.params.id;
-
-    // find pokemon by id from the pokedex json file
-    // (note: find() is a built-in method of JavaScript arrays)
-    let pokemon = obj.pokemon.find((currentPokemon) => {
-      return currentPokemon.id === parseInt(inputId, 10);
-    });
 
-    if (pokemon === undefined) {
+app.use(express.static('public'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({
+  extended: true
+}))
 
-      // send 404 back
-      response.status(404);
-      response.send("not found");
-    } else {
 
-      response.send(pokemon);
-    }
-  });
+//View Pokemon
+app.get('/', (request, response) => {
+    jsonfile.readFile('pokedex.json', (err, obj) => {
+        const pokemon = obj.pokemon;
+        const props = {
+            getpkm : pokemon
+        };
+        response.render('getpkm', props)
+    });
 });
 
-app.get('/', (request, response) => {
-  response.send("yay");
+
+//Intercept /pokemon/new with a form
+app.get('/pokemon/new', (request, response) => {
+  response.render('newform');
+});
+// add new pokemon
+app.post('/pokemon', (request, response) => {
+  console.log('Created New Pokemon')
+  let data = request.body;
+  let id = parseInt(data['id']);
+  data['id'] = id;
+  jsonfile.readFile(FILE, (err, obj) => {
+    obj.pokemon.push(data);
+    console.log(data);
+    response.send("New Pokemon Added!");
+    jsonfile.writeFile(FILE, obj);
+  })
 });
 
-/**
- * ===================================
- * Listen to requests on port 3000
- * ===================================
- */
-app.listen(3000, () => console.log('~~~ Tuning in to the waves of port 3000 ~~~'));
+
+
+//edit Pokemon Form
+app.get('/:id/edit', (request, response) => {
+  let index = parseInt(request.params.id);
+  response.render('editform');
+})
+//edit Pokemon
+app.put('/:id/edit', (request, response) => {
+  let editPokemon = request.body;
+  let id = parseInt(editPokemon['id']);
+  editPokemon['id'] = id;
+  jsonfile.readFile(FILE, (err, obj) => {
+    let pkm = obj.pokemon;
+    let pokemonReplace = pkm.find((currentPokemon) => {
+      return currentPokemon.id === editPokemon.id
+    })
+  let indexToReplace = pkm.indexOf(pokemonReplace);
+  pkm.splice(indexToReplace, 1, editPokemon);
```

Comment:
Because you use the splice command. Look it up.
I think what you're doing adds to the arrays.